PROVESFlightControllerReference/Components/TcSecurityDeframer/TcSecurityDeframer.cpp

```
@@ -54,11 +77,23 @@ void TcSecurityDeframer ::dataIn_handler(FwIndexType portNum, Fw::Buffer& data,
 
     {
-        Os::ScopeLock lock(this->m_sequenceNumberLock);
+        // Lock order: key store lock, then sequence number lock. Every other handler that takes
+        // both locks (none currently do) must follow the same order to avoid deadlock.
+        Os::ScopeLock keyLock(keyStoreLock());
+        Os::ScopeLock seqLock(this->m_sequenceNumberLock);
 
         // --- Validate SPI and anti-replay sequence number ---
-        const PacketValidator::Status validationStatus =
-            validatePacket(parseResult.securityHeader, this->m_sequenceNumber, this->m_sequenceNumberWindow);
+        PacketValidator::Status validationStatus = validatePacket(parseResult.securityHeader, this->m_sequenceNumber,
```

Clang (14.0.6)

[warning] variable 'validationStatus' is not initialized

(cppcoreguidelines-init-variables)